test/controllers/hosts_controller_test.rb

```diff
@@ -1652,40 +1652,64 @@ class Host::Test < Host::Base; end
   end
 
   test 'show power status for a host' do
+    expected_resp = {
+      "id" => @host.id,
+      "state" => "on",
+      "title" => "On"
+    }
+
     Host.any_instance.stubs(:supports_power?).returns(true)
     Host.any_instance.stubs(:supports_power_and_running?).returns(true)
     get :get_power_state, params: { :id => @host.id }, session: set_session_user, xhr: true
     assert_response :success
     response = JSON.parse @response.body
-    assert_equal({"id" => @host.id, "state" => "on", "title" => "On"}, response)
+    assert_equal(expected_resp.sort, response.sort)
   end
 
   test 'show power status for a powered off host' do
+    expected_resp = {
+      "id" => @host.id,
+      "state" => "off",
+      "title" => "Off"
+    }
+
     Host.any_instance.stubs(:supports_power?).returns(true)
     Host.any_instance.stubs(:supports_power_and_running?).returns(false)
     get :get_power_state, params: { :id => @host.id }, session: set_session_user, xhr: true
     assert_response :success
     response = JSON.parse @response.body
-    assert_equal({"id" => @host.id, "state" => "off", "title" => "Off"}, response)
+    assert_equal(expected_resp.sort, response.sort)
   end
 
   test 'show power status for a host that has no power' do
+    expected_resp = {
+      "id" => @host.id,
+      "state" => "na",
+      "title" => 'N/A',
+      "statusText" => "Power operations are not enabled on this host."
+    }
+
     Host.any_instance.stubs(:supports_power?).returns(false)
     get :get_power_state, params: { :id => @host.id }, session: set_session_user, xhr: true
     assert_response :success
     response = JSON.parse @response.body
-    assert_equal({"id" => @host.id, "state" => "na", "title" => 'N/A',
-      "statusText" => "Power operations are not enabled on this host."}, response)
+    assert_equal(expected_resp.sort, response.sort)
   end
 
   test 'show power status for a host that has an exception' do
+    expected_resp = {
+      "id" => @host.id,
+      "state" => "na",
+      "title" => "N/A",
+      "statusText" => "Failed to fetch power status: ERF42-9958 [Foreman::Exception]: Unknown power management support - can't continue"
+    }
+
     Host.any_instance.stubs(:supports_power?).returns(true)
     Host.any_instance.stubs(:power).raises(::Foreman::Exception.new(N_("Unknown power management support - can't continue")))
     get :get_power_state, params: { :id => @host.id }, session: set_session_user, xhr: true
     assert_response :success
     response = JSON.parse @response.body
-    assert_equal({"id" => @host.id, "state" => "na", "title" => "N/A",
-      "statusText" => "Failed to fetch power status: ERF42-9958 [Foreman::Exception]: Unknown power management support - can't continue"}, response)
+    assert_equal(expected_resp.sort, response.sort)
```

Comment:
Nit: you can remove the `()` here.

Reply:
well, that is a matter of code style taste ;)